1654531: Add proxy_scheme to rhsm.conf

cnsnyder opened the pull request:
This adds a new proxy_scheme option to rhsm.conf.
It allows users to specify what scheme should be used for the proxy url when writing repo definitions. It defaults to http (as that appears to be most common and will solve this bug).

wottop commented:
What is the outcome if they keep using the hostname with the protocol in it:
http://test.system.com ?

I like the separation, but are we going to see some headaches from people's current setups? Maybe the man page needs to spell this out for proxy_hostname.  You might want to respond with a message and exit if the proxy_hostname has a protocol designation on the start.

cnsnyder commented:
@wottop The current (and historical) norm is not to include the protocol in the hostname. For additional clarity I will add to the man page entry for hostname as well.

cnsnyder commented:
@wottop I've added some detail to the proxy_hostname section of the rhsm.conf.5 man page. In addition I've changed all reference of "proxy_protocol" back to "proxy_scheme" as per consensus with jsefler and khowell.

wottop commented:
ack
